vortex-ffi/src/file.rs

```diff
@@ -62,7 +63,9 @@ pub unsafe extern "C" fn File_open(options: *const FileOpenOptions) -> *mut FFIF
 
     assert!(!options.uri.is_null(), "File_open: null uri");
     let uri = CStr::from_ptr(options.uri).to_string_lossy();
-    let uri: Url = uri.parse().vortex_expect("File_open: parse uri");
+    let uri: Url = uri
+        .parse()
+        .vortex_expect_fn(|| format!("File_open: parse uri {uri}"));
```

Comment:
I'm not so keen on this panicking

Reply:
we have not decide on error handling yet, that is a bigger piece